### PR TITLE
Feature 3.16 patient pharmacist consultation cancellation

### PR DESCRIPTION
**Pull request by denisfruza**
https://trello.com/c/9Ymk7xuQ

### CODE REVIEW

File: backend/src/main/java/com/pharmacySystem/DTOs/AppointmentDTO.java [hunk 1]
```diff
@@ -3,9 +3,10 @@
 import com.pharmacySystem.model.appointment.Appointment;
 import com.pharmacySystem.model.appointment.AppointmentStatus;
 import com.pharmacySystem.model.appointment.AppointmentType;
+import com.pharmacySystem.model.appointment.CancelledAppointment;
 import com.pharmacySystem.model.workingHours.TimeInterval;
 
-public class AvailableAppointmentDTO {
+public class AppointmentDTO {
```
Comment (dbulaja98) [inline backend/src/main/java/com/pharmacySystem/DTOs/AppointmentDTO.java:9]: PLeeeeaaaaaaase change name :D

File: backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java [hunk 2]
```diff
@@ -185,12 +188,38 @@ private double findLowestAppointmentPriceByPharmacyId(Long pharmacyId, Set<Appoi
...
+	private void removeCancelledAppointments(Set<AppointmentDTO> availableAppointments, Long patientId) {
+		Set<AppointmentDTO> appointmentsToRemove = new HashSet<AppointmentDTO>();
+		for(AppointmentDTO available : availableAppointments) {
+			if(isAppointmentToRemove(available, patientId))
+				appointmentsToRemove.add(available);
+		}
+		availableAppointments.removeAll(appointmentsToRemove);	
+	}
+	
+	private boolean isAppointmentToRemove(AppointmentDTO appointment, Long patientId){
```
Comment (dbulaja98) [inline backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java:208]: IsAppointmentForRemoval?

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java (hunk 2 above)
Comment (denisfruza) [reply backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java:208]: fixed

File: backend/src/main/java/com/pharmacySystem/DTOs/AppointmentDTO.java (hunk 1 above)
Comment (denisfruza) [reply backend/src/main/java/com/pharmacySystem/DTOs/AppointmentDTO.java:9]: fixed

**Review by dbulaja98 — APPROVED** [review]

**Review by cakareli — APPROVED** [review]

**Pull request merged**